flake.nix

```diff
@@ -0,0 +1,98 @@
+{
+  description = "Souffle Datalog bindings for Haskell";
+  inputs = {
+    np.url = "github:nixos/nixpkgs?ref=haskell-updates";
+    fu.url = "github:numtide/flake-utils?ref=master";
+    hls.url = "github:haskell/haskell-language-server?ref=master";
+  };
+  outputs = { self, np, fu, hls }:
+    with fu.lib;
+    eachSystem [ "x86_64-linux" ] (system:
+      let
+        version = with np.lib;
+          "${substring 0 8 self.lastModifiedDate}.${self.shortRev or "dirty"}";
+        config = { };
+        mkSouffle = { p, gcc ? 10 }:
```

Comment:
Move this function to a separate souffle.nix file and import it here? Or is this not support by nix flakes?

Reply:
Ideally, we'd upstream this to nixpkgs, so we don't have to specify it here, but with each new souffle release we would lag behind immediately.